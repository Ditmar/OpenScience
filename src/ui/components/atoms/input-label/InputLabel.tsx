import React, { useEffect, useState } from 'react';
import { Box, MenuItem, Select, type SelectChangeEvent } from '@mui/material';
import type { IProps, Option } from './types/IProps';
// import styles from './InputCountry.module.scss';
import { Icon } from '../../../utils/svg-icons/icons';
import globe from '../../../../assets/icons/globe.svg?raw';
import close_circle from '../../../../assets/icons/close_circle.svg?raw';
import info from '../../../../assets/icons/info.svg?raw';
import { selectSx, textSx, iconSx } from './styles';
import { InputSelect } from 'ui/components/atoms/input-select/InputSelect';

export function InputLabel(props: IProps) {
      const { label, error, helperText, disabled, size } = props;
    const [country, setCountry] = useState('');
    const [keyboardFocus, setKeyboardFocus] = useState(false);

    const handleChange = (event: SelectChangeEvent) => setCountry(event.target.value);

    useEffect(() => {
        const onKey = () => setKeyboardFocus(true);
        const onMouse = () => setKeyboardFocus(false);
        window.addEventListener('keydown', onKey);
        window.addEventListener('mousedown', onMouse);
        return () => {
            window.removeEventListener('keydown', onKey);
            window.removeEventListener('mousedown', onMouse);
        };
    }, []);

      const labelStyles = textSx({ disabled, error, size });
      const helperStyles = textSx({ disabled, error, size });
      const closeStyles = iconSx({ disabled, error });
      const infoStyles = iconSx({ disabled, error });

    return (
        <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center', mb: 1 }}>
            <Box sx={{ display: 'flex', alignItems: 'center', gap: 0.5 }}>
                {globe && <Icon src={globe} styles={closeStyles as React.CSSProperties} />}
                <Box component="p" sx={labelStyles}>{label || ''}</Box>
            </Box>
            <Box>{close_circle && <Icon src={close_circle} styles={closeStyles as React.CSSProperties} />}</Box>
        </Box>
    );
}
