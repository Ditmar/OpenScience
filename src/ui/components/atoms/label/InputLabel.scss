@import '../../../../globals/variables';
@import '../../../../globals/mixins';
@import '../../../../globals/globals';

.input-label {
  color: $ads-icontextstat-color-title;
  display: block;
  font-size: $ads-paragraph-xs;
  font-weight: $ads-icontextstat-weight-title;
  margin-bottom: 8px;

  &--disabled {
    color: $ads-input-disabled-color;
    opacity: $ads-state-disabled-opacity;
  }
}
