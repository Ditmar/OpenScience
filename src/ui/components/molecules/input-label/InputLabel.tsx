import { Box } from '@mui/material';

import { IconMui } from '../../atoms/icon-mui/IconMui';
import { InputLabelText } from '../../atoms/input-label-text/InputLabelText';

import type { IProps } from './types/IProps';

export function InputLabel(props: IProps) {
  const {
    label,
    labelError,
    labelDisabled,
    labelHelper,
    labelSize,
    iconSize,
    leftIcon,
    leftIconError,
    leftIconDisabled,
    rightIcon,
    rightIconError,
    rightIconDisabled,
  } = props;

  return (
    <Box sx={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between' }}>
      <Box sx={{ display: 'flex', alignItems: 'center', gap: '0.5rem' }}>
        {leftIcon ? (
          <IconMui
            iconName={leftIcon}
            size={iconSize}
            error={leftIconError}
            disabled={leftIconDisabled}
          />
        ) : null}
        <InputLabelText
          label={label}
          error={labelError}
          disabled={labelDisabled}
          helper={labelHelper}
          size={labelSize}
        />
      </Box>
      <Box>
        {rightIcon ? (
          <IconMui
            iconName={rightIcon}
            size={iconSize}
            error={rightIconError}
            disabled={rightIconDisabled}
          />
        ) : null}
      </Box>
    </Box>
  );
}
