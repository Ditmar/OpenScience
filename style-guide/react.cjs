const { resolve } = require('node:path');
const project = resolve(process.cwd(), 'tsconfig.json');

module.exports = {
  env: {
    browser: true,
    es2021: true,
    es6: true,
    jest: true,
  },
  extends: [
    require.resolve('./rules/base'),
    require.resolve('./configurations/_base'),
    'plugin:react-hooks/recommended',
    'airbnb',
    'plugin:storybook/recommended',
    require.resolve('./configurations/typescript'),
    require.resolve('./configurations/vitest'),
  ],
  parserOptions: {
    project,
    babelOptions: {
      presets: ['@babel/preset-react'],
    },
  },
  plugins: ['react-refresh'],
  settings: {
    react: {
      version: 'detect',
    },
  },
  globals: {
    JSX: true,
  },
  rules: {
    'react/react-in-jsx-scope': 'off',
    'react/button-has-type': 'off',
    'react-refresh/only-export-components': [
      'warn',
      { allowExportNames: ['meta', 'links', 'headers', 'loader', 'action'] },
    ],
  },
};
