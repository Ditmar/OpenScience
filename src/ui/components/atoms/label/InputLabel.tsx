import { Typography } from '@mui/material';
import React from 'react';
import type { InputLabelProps } from './types/IProps';
import './InputLabel.scss';

export function InputLabel({ htmlFor, children }: InputLabelProps) {
  return (
    <Typography variant="subtitle2" htmlFor={htmlFor} component="label">
      {children}
    </Typography>
  );
}
