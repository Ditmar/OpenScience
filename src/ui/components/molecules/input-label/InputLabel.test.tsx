import { render } from '@testing-library/react';
import { InputLabel } from './InputLabel';

describe('InputLabel Component', () => {
  it('renders disabled state', () => {
    const { container } = render(<InputLabel label="Input Label" />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with icons', () => {
    const { container } = render(
      <InputLabel
        label="Input Label"
        leftIcon="LanguageSharpIcon"
        rightIcon="CancelOutlinedIcon"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with error', () => {
    const { container } = render(
      <InputLabel label="Input Label" labelError leftIconError rightIconError />,
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot disabled', () => {
    const { container } = render(
      <InputLabel label="Input Label" labelDisabled leftIconDisabled rightIconDisabled />,
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot helper', () => {
    const { container } = render(<InputLabel label="Input Label" labelHelper />);
    expect(container).toMatchSnapshot();
  });
});
