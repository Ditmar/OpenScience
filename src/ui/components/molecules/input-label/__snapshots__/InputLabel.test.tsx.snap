// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputLabel Component > matches snapshot disabled 1`] = `
<div>
  <div
    class="MuiBox-root css-1lekzkb"
  >
    <div
      class="MuiBox-root css-19nzcwv"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1b4s42d-MuiTypography-root"
      >
        Input Label
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    />
  </div>
</div>
`;

exports[`InputLabel Component > matches snapshot helper 1`] = `
<div>
  <div
    class="MuiBox-root css-1lekzkb"
  >
    <div
      class="MuiBox-root css-19nzcwv"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-9lmazj-MuiTypography-root"
      >
        Input Label
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    />
  </div>
</div>
`;

exports[`InputLabel Component > matches snapshot with error 1`] = `
<div>
  <div
    class="MuiBox-root css-1lekzkb"
  >
    <div
      class="MuiBox-root css-19nzcwv"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ek0aq9-MuiTypography-root"
      >
        Input Label
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    />
  </div>
</div>
`;

exports[`InputLabel Component > matches snapshot with icons 1`] = `
<div>
  <div
    class="MuiBox-root css-1lekzkb"
  >
    <div
      class="MuiBox-root css-19nzcwv"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-16vjuwa-MuiSvgIcon-root"
        data-testid="LanguageSharpIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2s.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2s.07-1.35.16-2h4.68c.09.65.16 1.32.16 2s-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2s-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eq936y-MuiTypography-root"
      >
        Input Label
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-16vjuwa-MuiSvgIcon-root"
        data-testid="CancelOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m3.59-13L12 10.59 8.41 7 7 8.41 10.59 12 7 15.59 8.41 17 12 13.41 15.59 17 17 15.59 13.41 12 17 8.41z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`InputLabel Component > renders disabled state 1`] = `
<div>
  <div
    class="MuiBox-root css-1lekzkb"
  >
    <div
      class="MuiBox-root css-19nzcwv"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-eq936y-MuiTypography-root"
      >
        Input Label
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    />
  </div>
</div>
`;
